hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -137,6 +137,10 @@ public List<DeletedBlocksTransaction> getFailedTransactions(int count,
             }
           }
         } else {
+          iter.seek(startTxId);
+          if (iter.hasNext()) {
+            iter.next();
```

Comment:
The problem is if we don't do next here then `nextTx` still points to old location. And when we do [getvalue](https://github.com/apache/ozone/blob/6b40003bb251a4ad699f38e6cc3adc6a354adaed/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java#L141) it returns the old value instead the value where we seek.

Reply:
at line 145, we do next() -- `iter.next().getValue();` do still have issue?

Reply:
Yes `iter.next().getValue()` returns old value at L145 because [nextTx](https://github.com/apache/ozone/blob/a8db9cd4766db9086b6c28d431e2faf1df1d9a49/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java#L124) is not updated in `seek.`

Reply:
Changed seek to update nextTx value.